File: 102.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function (root) {
  if (!root) {
    return [];
  }
  const res = [];
  let level = [];
  let nextLevel = [root];

  while (nextLevel.length) {
    level = nextLevel;
    nextLevel = [];
    res.push([]);

    for (node of level) {
      res.at(-1).push(node.val);
      if (node.left) nextLevel.push(node.left);
      if (node.right) nextLevel.push(node.right);
    }
  }

  return res;
};

File: 102.go

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func levelOrder(root *TreeNode) [][]int {
	if root == nil {
		return [][]int{}
	}

	var res [][]int
	var level []*TreeNode
	var nextLevel = []*TreeNode{root}

	for len(nextLevel) > 0 {
		level = nextLevel
		nextLevel = []*TreeNode{}
		res = append(res, []int{})

		for _, node := range level {
			res[len(res)-1] = append(res[len(res)-1], node.Val)
			if node.Left != nil {
				nextLevel = append(nextLevel, node.Left)
			}
			if node.Right != nil {
				nextLevel = append(nextLevel, node.Right)
			}
		}
	}

	return res
}
